analyzers/src/SonarAnalyzer.CSharp/Rules/ClassShouldNotBeAbstract.cs

```diff
@@ -26,105 +26,113 @@
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using SonarAnalyzer.Helpers;
+using StyleCop.Analyzers.Lightup;
 
 namespace SonarAnalyzer.Rules.CSharp
 {
     [DiagnosticAnalyzer(LanguageNames.CSharp)]
     [Rule(DiagnosticId)]
     public sealed class ClassShouldNotBeAbstract : SonarDiagnosticAnalyzer
     {
-        internal const string DiagnosticId = "S1694";
-        private const string MessageFormat = "Convert this 'abstract' class to {0}.";
-        internal const string MessageToInterface = "an interface";
-        internal const string MessageToConcreteClass = "a concrete class with a protected constructor";
+        private const string DiagnosticId = "S1694";
+        private const string MessageFormat = "Convert this 'abstract' {0} to {1}.";
+        private const string MessageToInterface = "an interface";
+        private const string MessageToConcreteImplementation = "a concrete type with a protected constructor";
 
-        private static readonly DiagnosticDescriptor rule =
+        private static readonly DiagnosticDescriptor Rule =
             DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(Rule);
 
-        protected override void Initialize(SonarAnalysisContext context)
-        {
+        protected override void Initialize(SonarAnalysisContext context) =>
             context.RegisterSymbolAction(
                 c =>
                 {
-                    var symbol = c.Symbol as INamedTypeSymbol;
-                    if (!symbol.IsClass() ||
-                        !symbol.IsAbstract ||
-                        ClassHasInheritedAbstractMembers(symbol))
+                    var symbol = (INamedTypeSymbol)c.Symbol;
+                    if (symbol.Kind != SymbolKind.NamedType
```

Comment:
I think this check is uncovered as it is redundant. Is it possible to get a NamedTypeSymbol which is not of symbolKind namedtype?